File: missions/missions_solution.js
function solution(D, X) {
    let stack = [D[0]];
    let counter = 0;
    let min = D[0], max = D[0];
    for (let i = 0; i < D.length; i++) {
        if (Math.abs(D[i] - min) <= X && Math.abs(D[i] - max) <= X) {
            stack.push(A[i]);
            if (D[i] < min) {
                min = D[i];
            }
            if (D[i] > max) {
                max = D[i];
            }
        } else {
            counter++;
            stack = [D[i]];
            min = D[i];
            max = D[i];
        }
    }
    if (stack.length !== 0) counter++;
    return counter;
}

File: missions/missions_solution.py
def solution(D, X):
    stack = [D[0]]
    counter = 0
    min_val = D[0]
    max_val = D[0]

    for i in range(len(D)):
        if abs(D[i] - min_val) <= X and abs(D[i] - max_val) <= X:
            stack.append(D[i])
            min_val = min(D[i], min_val)
            max_val = max(D[i], max_val)
        else:
            counter += 1
            stack = [D[i]]
            min_val = D[i]
            max_val = D[i]

    if stack:
        counter += 1

    return counter
